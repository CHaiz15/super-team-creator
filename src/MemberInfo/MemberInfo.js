import React from 'react'
import './MemberInfo.scss'

export const MemberInfo = (props) => {
  let {member} = props.location.state

  return (
    <section className='member-info'>
      <section className='info-section'>
          <div className='names-div'>
            <h1 className='name-title'>{member.name}</h1>
          </div>
        <div className='info-holder'>
          <img className='selected-member-image' src={member.image.url} alt='member one'/>
        <div className='stats-container'>
          <ul className='powers-list'>Appearance:
            <li>Gender: {member.appearance.gender}</li>
            <li>Race: {member.appearance.race}</li>
            <li>Height: {member.appearance.height[0]}</li>
            <li>Weight: {member.appearance.weight[0]}</li>
            <li>Eye Color: {member.appearance['eye-color']}</li>
            <li>Hair Color: {member.appearance['hair-color']}</li>
          </ul>
          <ul className='powers-list'>Biography:
            <li>Real Name: {member.biography['full-name']}</li>
            <li>Home: {member.biography['place-of-birth']}</li>
            <li>Work: {member.work.occupation}</li>
            <li>Alignment: {member.biography.alignment}</li>
            <li>Publisher: {member.biography.publisher}</li>
            <ul>Aliases: {member.biography.aliases.map(alias => <li>"{alias}"</li>)}</ul>
          </ul>
          <ul className='powers-list'>Stats:
            <li>Intelligence: {member.powerstats.intelligence}</li>
            <li>strength: {member.powerstats.strength}</li>
            <li>Speed: {member.powerstats.speed}</li>
            <li>Durability: {member.powerstats.durability}</li>
            <li>Power: {member.powerstats.power}</li>
            <li>Combat: {member.powerstats.combat}</li>
          </ul>
        </div>
        </div>
      </section>
    </section>
  )
}

export default MemberInfo