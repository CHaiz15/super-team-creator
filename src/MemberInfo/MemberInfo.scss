.member-info {  
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.info-section {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  height: 90vh;
  width: 90vw;
  background-color: white;
  opacity: 0.95;
  border-radius: 10px;
}

.info-holder {
  text-align: center;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  height: 80%;
  width: 80%;
}

.selected-member-image {
  margin: -5%;
  height: 40%;
  width: 30%;
  border-radius: 50%;
}

.names-div {
  display: flex;
  flex-direction: row;
  margin-top: -5%;
}

.name-title {
  display: flex;
  font-size: 5em;
}

.stats-container {
  margin: 10% 0 0 0;
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-around;
}
