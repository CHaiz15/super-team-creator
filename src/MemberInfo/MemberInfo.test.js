import React from 'react';
import { MemberInfo } from './MemberInfo.js';
import { shallow } from 'enzyme'

describe('MemberInfo', () => {
  let wrapper;
  let expectedProps;
  beforeEach(() => {
    expectedProps = {
      location: {
      state: { 
        member: {
          name: "Hulk",
          powerstats: {
              intelligence: "88",
              strength: "100",
              speed: "63",
              durability: "100",
              power: "98",
              combat: "85"
          },
          biography: {
              "full-name": "Bruce Banner",
              "alter-egos": "No alter egos found.",
              aliases: [
                  "Annihilator",
                  "Captain Universe",
                  "Joe Fixit",
                  "Mr. Fixit",
                  "Mechano",
                  "Professor",
                  "Jade Jaws",
                  "Golly Green Giant"
              ],
              "place-of-birth": "Dayton, Ohio",
              "first-appearance": "Incredible Hulk #1 (May, 1962)",
              publisher: "Marvel Comics",
              alignment: "good"
          },
          appearance: {
              gender: "Male",
              race: "Human / Radiation",
              height: [
                  "8'0",
                  "244 cm"
              ],
              weight: [
                  "1400 lb",
                  "630 kg"
              ],
              "eye-color": "Green",
              "hair-color": "Green"
          },
          image: {
              url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"
          }
        },
      }
    }
  }
    wrapper = shallow(<MemberInfo {...expectedProps} />);
  })  
  it('should be an instance of the MemberInfo component', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
