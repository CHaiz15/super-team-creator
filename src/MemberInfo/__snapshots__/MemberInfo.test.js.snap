// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberInfo should be an instance of the MemberInfo component 1`] = `ShallowWrapper {}`;
